core/Updates/4.0.0-b1.php

```diff
@@ -95,6 +95,27 @@ public function getMigrations(Updater $updater)
             $migrations[] = $this->migration->plugin->activate('CustomJsTracker');
         }
 
+        // Prepare all installed tables for utf8mb4 conversions. e.g. make some indexed fields smaller so they don't exceed the maximum key length
+        $allTables = DbHelper::getTablesInstalled();
+
+        $migrations[] = $this->migration->db->changeColumnType('session', 'id', 'VARCHAR(191)');
+        $migrations[] = $this->migration->db->changeColumnType('site_url', 'url', 'VARCHAR(190)');
+        $migrations[] = $this->migration->db->changeColumnType('option', 'option_name', 'VARCHAR(191)');
+
+        foreach ($allTables as $table) {
+            if (preg_match('/archive_/', $table) == 1) {
+                $tableNameUnprefixed = Common::unprefixTable($table);
+                $migrations[] = $this->migration->db->changeColumnType($tableNameUnprefixed, 'name', 'VARCHAR(190)');
+            }
+        }
+
+        // Convert all tables to utf8mb4 if it is available
+        if ('utf8mb4' === DbHelper::getDefaultCharset()) {
```

Comment:
as mentioned for Matomo for WordPress I'd kind of need a way to not execute this unless the user actually uses that charset in WP. 

Also some people will maybe not want to upgrade to utf8mb4 for some reason so we need to have a way to force utf8 (or some other charset) in that method.